ISPy/util/neural_estimation.py

```diff
@@ -0,0 +1,631 @@
+        for ix in tqdm(range(nx)):
+            for iy in range(ny):
+                temp = np.interp(original_logtau, logtau, np.abs(prediction[ix, iy, 0, :]))
+                vlos = np.interp(original_logtau, logtau, prediction[ix, iy, 1, :])
+                vturb = np.interp(original_logtau, logtau, np.abs(prediction[ix, iy, 2, :]))
+                Bln = np.interp(original_logtau, logtau, prediction[ix, iy, 3, :])
+                Bho = np.interp(original_logtau, logtau, np.abs(prediction[ix, iy, 4, :]))
+                Bazi = np.interp(original_logtau, logtau, prediction[ix, iy, 5, :])
+
+                m.ltau[0, iy, ix, :] = original_logtau
+                m.temp[0, iy, ix, :] = temp * 1e3
+                m.vlos[0, iy, ix, :] = vlos * 1e5
+                m.pgas[0, iy, ix, :] = 1.0
+                m.vturb[0, iy, ix, :] = vturb * 1e5
+                m.Bln[0, iy, ix, :] = Bln * 1e3
+                m.Bho[0, iy, ix, :] = Bho * 1e3
+                m.azi[0, iy, ix, :] = Bazi
+
+        # Write the model
+        m.write(nameoutput)
+
+    def prepare_train(self, model_train_list, stokes_train_list, logtau_train_list):
+        """Creates a dataset for training
+
+        Parameters
+        ----------
+        model_train_list : list of strings
+            List of models in STiC format used for training
+        stokes_train_list : list of strings
+            List of observed or synthetic profiles for training
+        logtau_train_list : list
+            List of logtau values included in the training
+        """
+        self.logtau = np.array(logtau_train_list)
+
+        stokelist, cubelist = [], []
+        for simu in range(len(model_train_list)):
+            m = sp.model(model_train_list[simu])
+            s = sp.profile(stokes_train_list[simu])
+            idx = np.where(s.weights[:, 0] < 1.0)[0]
+            indices = sorted(findindex(self.logtau, m.ltau[0, 0, 0, :]))
+            ni = len(indices)
+
+            # Physical parameters
+            supercube = np.zeros((ni * self.num_params, m.temp.shape[1], m.temp.shape[2]))
+            supercube[:ni] = m.temp[0, :, :, indices] / 1e3
+            supercube[ni:2 * ni] = m.vlos[0, :, :, indices] / 1e5
+            supercube[ni * 2:3 * ni] = m.vturb[0, :, :, indices] / 1e5
+            supercube[ni * 3:4 * ni] = m.Bln[0, :, :, indices] / 1e3
+            supercube[ni * 4:5 * ni] = m.Bho[0, :, :, indices] / 1e3
+            supercube[ni * 5:6 * ni] = m.azi[0, :, :, indices]
+
+            # Stokes parameters
+            stokes = np.concatenate([s.dat[0, :, :, idx, 0], 1e0 * s.dat[0, :, :, idx, 1],
+                                     1e0 * s.dat[0, :, :, idx, 2], 1e0 * s.dat[0, :, :, idx, 3]])
+
+            stokelist.append(stokes)
+            cubelist.append(supercube)
+
+        self.cubelist = cubelist
+        self.stokelist = stokelist
+        self.nl = len(stokes)
+
+        
+    def training(self, name='network1', option='start', nepochs=20, extranoise=5e-4,
```

Comment:
Simplify name to `train` if the method on line 527 is renamed?

Reply:
Fixed